samza-api/src/main/java/org/apache/samza/operators/data/IncomingSystemMessage.java

```diff
@@ -23,54 +23,47 @@
 
 
 /**
- * This class implements a {@link Message} that encapsulates an {@link IncomingMessageEnvelope} from the system
- *
+ * A {@link Message} implementation that provides additional information about its input {@link SystemStreamPartition}
+ * and its {@link Offset} within it.
```

Comment:
1. /s/its input and its offset within it/ its input and its offset. Do you think `within it` is probably redundant?